static/js/types.ts

```diff
@@ -0,0 +1,22 @@
+// TODO: Move this to message_store when it is
+// converted to TypeScript.
+interface BaseMessage {
+    content: string;
+    match_content?: string;
+    match_subject?: string;
+    sender_email: string;
+}
+
+export interface PrivateMessage extends BaseMessage {
+    type: "private";
+    to_user_ids: string;
+}
+
+export interface StreamMessage extends BaseMessage {
+    type: "stream";
+    stream_id: number;
+    subject?: string;
+    topic: string;
```

Comment:
It seems to me that `subject` is the required one and `topic` only exists if we’ve run `convert_message_topic`?

We really should finish this migration, or at least come up with a better way to deal with the intermediate state. Perhaps we could `export const TOPIC_NAME = "subject" as const` somewhere and use `message[TOPIC_NAME]`, like we do in the backend, rather than putting up with pervasive uncertainty about what our fields are named.

Reply:
Yeah, agreed -- I don't remember what's left with that migration, but maybe worth starting a thread on what we need to do to finish it.

Reply:
Does the `util: Don't depend on msg.topic to be defined.` commit look good? I can also try to attempt to do this for all the modules in a separate PR before we integrate this if needed or we can gradually do this change as we convert modules.

Reply:
The idea with `TOPIC_NAME` would be to remove `convert_message_topic` and all of the conditional logic that tries to detect which of `topic` or `subject` is present at runtime. It would just always be `subject` and always be accessed via `[TOPIC_NAME]`. (Except, I guess, in locally stored drafts, where we’re already doomed.)

Reply:
Yeah, I got the idea already, it can't be done here though because I think other places still pass message objects where the `topic` property is updated instead of the `subject` (not fully sure)? I will do another PR for this then.